media/js/mozorg/contribute/taskview.js

```diff
@@ -0,0 +1,273 @@
+$(function() {
+    'use strict';
+
+    var $taskSteps = $('.task-steps');
+    var $stepOne = $('#step_one');
+    var $thankYou = $('#thankyou');
+    var $getFirefox = $('.get-firefox');
+    var $actionButton = $('.action');
+    var $downloadButton = $('.download-link');
+
+    var $followButton = $('.follow-mozilla')
+    var intentURL = $followButton.attr('href');
+
+    var $signupTweetForm = $('#signup-tweet');
+
+    // some tasks, like installing Whimsy, required the user to be using Firefox
+    if ($getFirefox.length > -1 && !Mozilla.Client.isFirefox) {
+        $getFirefox.toggleClass('hidden');
+        // hide any action buttons that forms part of the task.
+        $actionButton.addClass('hidden');
+    }
+
+    /**
+     * Toggles the completed class on the relevant step, and calls
+     * taskComplete() if this was the final step of the task.
+     */
+    function completeStep($step) {
+        var $stepOne = $('#step_one');
+        var $stepTwo = $('#step_two');
+
+        if ($step.data('step') === 'one') {
+            $stepOne.toggleClass('completed');
+        } else if ($step.data('step') === 'two') {
+            $stepTwo.toggleClass('completed');
+        }
+
+        if ($step.data('complete') === true) {
+            taskComplete();
+        }
+    }
+
+    /**
+     * Called once all steps of the task has been completed. This will
+     * show the thank message and scroll it into view.
+     */
+    function taskComplete() {
+        $thankYou.removeClass('visibly-hidden');
+        $thankYou.attr('aria-hidden', 'false');
+        $thankYou[0].scrollIntoView();
+        $thankYou.focus();
+    }
+
+    /**
+     * Waits for the initial tab/window to become visible, and then
+     * proceeds to complete the relevant task step.
+     */
+    function handleVisibilityChange($step) {
+        $(document).on('visibilitychange.taskview', function() {
```

Comment:
Hmmm yes, so IE9 and lower does not support this so yes agreed. The thing is then, without having tested, that means that if someone in IE(unlikely but possible) completes any task that relies on this, the will not see the completion step so, perhaps there needs to be a fallback here for those browsers anyhow.

Will think about this some. Thx @alexgibson